src/components/tabs/tabs.js

```diff
@@ -323,37 +362,53 @@ class Tabs extends React.Component {
       {
         'carbon-tabs__headers__header--error': tabHasError,
         'carbon-tabs__headers__header--warning': tabHasWarning,
-        'carbon-tabs__headers__header--selected': tab.props.tabId === this.state.selectedTabId
+        'carbon-tabs__headers__header--selected': this.isTabSelected(tab.props.tabId)
       }
     );
   }
 
+  isTabSelected = tabId => tabId === this.state.selectedTabId;
+
+  get children() {
+    return compact(React.Children.toArray(this.props.children));
+  }
+
+  get tabIds() {
+    return this.children.map(child => child.props.tabId);
+  }
```

Comment:
can you add comment/descriptions to these new methods please?

Reply:
Done